prog1.py

```diff
@@ -1,28 +1,66 @@
+#!/usr/bin/python
+
+import argparse
 import psutil
 import subprocess
 import re
 import time
+import multiprocessing
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-n", "--number", type=int, help="Number of files to create")
+parser.add_argument("-s", "--size", type=int, help="Size of file to create in MB")
+parser.add_argument("-m", dest='min_free', type=int, help="Minimum disk free space in MB")
+parser.add_argument("-l", dest="list",  action='store_true', help="List available mount points. Should be used with \'-m\' option")
+parser.add_argument("-p", dest='mount_point', type=str, help="Mount point to store files")
 
-file_num = 3
-file_size = 4096
-#free_space = 
+args = parser.parse_args()
 
+file_num = args.number
+file_size = args.size
+min_free = args.min_free 
+mount_point = args.mount_point
 
 # Detect local mounted disk
-local_disk_mounts = []
-partitions = psutil.disk_partitions(all=False)
-pattern = re.compile(".*/dev/[h,s]d[a-z].*")
-for p in partitions:
-   if pattern.match(str(p)) and psutil.disk_usage(p.mountpoint).free > 1024:
-        local_disk_mounts.append(p.mountpoint)
+def local_disk_mounts(min_free):
+    local_disk_mounts = []
+    partitions = psutil.disk_partitions(all=False)
+    pattern = re.compile(".*/dev/(mapper|[h,s]d[a-z]).*")
+    for p in partitions:
+       # psutil.disk_usage returns values in bytes
+       if pattern.match(str(p)) and psutil.disk_usage(p.mountpoint).free > (min_free * 1024 * 1024):
+            local_disk_mounts.append(p.mountpoint)
+    return local_disk_mounts
 
 def run_dd(num, size, mount):
-    for n in range(num):
-        subprocess.check_output(['dd', 'if=/dev/zero', 'of=%sfile%s.txt' % (mount, n), 'count=1', 'bs=%s' % size ], stderr=subprocess.STDOUT)
+    try:
+        subprocess.check_output(['dd', 'if=/dev/zero', 'of=%sfile%s.txt' % (mount, num), 'status=none', 'count=1', 'bs=%s' % (size * 1024 * 1024)])
+        print('File \'%sfile%s.txt\' created succefully! Size: %s MB' % (mount, num, size))
+    except subprocess.CalledProcessError as e:
+        print e.output
+
+def list_disks():
+    print "Available mounted disks:"
+    print '\n'.join([i for i in local_disk_mounts(min_free)])
 
+if __name__ == '__main__':
+    start = time.time()
+    if args.list and min_free:
+        list_disks()
+        exit()
+    elif args.list and not min_free:
+        print "Option \'-l\' cant be use\'d without \'-m\' option!"
+        exit()
 
-start = time.time()
-run_dd(file_num, file_size, local_disk_mounts[0])
-end = time.time()
+    if args.number and args.size and args.mount_point:
```

Comment:
1. The program needs to find a filesystem on the local disk. No need to specify mount point explicitly.
2. In case no free partitions (FS-es) are found -- you need to inform the user about this. Currently you simply ignores empty list returned by `list_disk_mounts()` function  as far as I See